src/vectors/angular.rs

```diff
@@ -35,6 +39,9 @@ use crate::{
 ///
 /// * [Cosine similarity](https://en.wikipedia.org/wiki/Cosine_similarity)
 pub fn cosine<T: Number, U: Float>(x: &[T], y: &[T]) -> U {
+    debug_assert_eq!(x.len(), y.len());
+    debug_assert!(!x.is_empty());
+
```

Comment:
I am no longer of the opinion that we want the debug asserts here because I recently came across a use case at work where the inputs would have different lengths. Let's discuss this more during the meeting.